### PR TITLE
Add Functionality to change the power state of a cluster

### PR DESCRIPTION
**Pull request by starhawking**
Added Functionality to simplify changing the power state of a user's cluster.

### CODE REVIEW

File: library/vsphere_vm [hunk 1]
```diff
@@ -6,7 +6,26 @@ from pyVmomi import vim, vmodl
 from pyVim import connect
 
 PYTHON_IS_MODERN = sys.version_info >= (2, 7, 9)
-
+powerStates = {
```
Comment (ianunruh) [inline library/vsphere_vm]: Use constants for these in underscore uppercase (e.g. POWER_STATES) and single quotes everywhere

File: library/vsphere_vm [hunk 2]
```diff
@@ -6,7 +6,26 @@ from pyVmomi import vim, vmodl
...
+    "poweredOn":  vim.VirtualMachine.PowerState.poweredOn,
+    "poweredOff": vim.VirtualMachine.PowerState.poweredOff,
+    "suspended":   vim.VirtualMachine.PowerState.poweredOff
+    }
+
+changePowerState = {
+    "shutdown": powerStates["poweredOff"],
+    "start"   : powerStates["poweredOn"],
+    "reboot"  : powerStates["poweredOff"] ,
+    "suspend"  : powerStates["suspended"],
```
Comment (ianunruh) [inline library/vsphere_vm]: Don't try to line up the colons, just use natural spacing

File: library/vsphere_vm [hunk 3]
```diff
@@ -6,7 +6,26 @@ from pyVmomi import vim, vmodl
...
+    }
+vmTasks = {
+    "shutdown" : "PowerOffVM_Task",
+    "start"    : "PowerOnVM_Task",
+    "reboot"  : "ResetVM_Task",
+    "suspend"  : "SuspendVM_Task",
+    }
+    
+validAnsibleStates=['present', 'absent']+changePowerState.keys()
```
Comment (ianunruh) [inline library/vsphere_vm]: Spacing between all the operators here

**Comment by ianunruh** [comment]
Added some comments on styling. Commits need to be squashed into one, using present tense in the commit subject (e.g., Add cluster power state management)

**Comment by starhawking** [comment]
I've updated my branch with the changes mentioned.

**Pull request merged**